agent.go

```diff
@@ -1214,10 +1214,19 @@ func addMounts(config *configs.Config, fsmaps []hyper.Fsmap) error {
 	for _, fsmap := range fsmaps {
 
 		source := fsmap.Source
+		var err error
+
 		if !fsmap.AbsolutePath {
 			source = filepath.Join(mountShareDirDest, fsmap.Source)
 		}
 
+		if fsmap.SCSIAddr != "" {
+			source, err = getSCSIDisk(fsmap.SCSIAddr)
```

Comment:
- This is resetting `source` so is it worth adding an extra check to see if `source != ""` here? Or is it guaranteed that if `SCSIAddr` is set that `Source` is irrelevant?

  Could you maybe add some comments to the `Fsmap` type to explain this?

- Does this check need to come before the `AbsolutePath` check above as currently that code is being executed for no reason if we're using a SCSI device?

Reply:
I have added in comments